### PR TITLE
feat(metrics): expose dogfood failure root-cause clusters

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_48b6ca97ac0b4aa9aab44125` (agent: `gemini`).

### Task
Improve AWF observability by exposing root-cause clusters for recent dogfood failures. Problem: recent workspaces failed for different reasons: stale coverage baseline, bad Gemini model id, OpenCode-generated syntax errors, missing managed worktree during fix loop, GitHub auth/PR creation failures, and validation failures. The console/API should make these patterns visible instead of showing only generic agent_failure or validation_failure. Expected behavior: extend the existing failure metrics/summary surface with a backward-compatible root_cause_clusters field. Cluster by agent, agent_model when present, failure_reason, likely phase/root cause, and actionable next action. Initial signal detection must include AGENT_AUTH_FAILED, model not found or 404, missing worktree, coverage threshold failure, syntax/import errors during validation, GitHub transient/auth errors, and unknown validation failure. Include counts and sample workspace IDs per cluster. Keep API backward-compatible. TDD: add focused tests first for mixed failed workspace rows, agent_model extraction from task_policy, empty history, API route serialization, and existing failure_groups/latest_examples remaining populated unchanged. Do not run repo-wide formatters. Do not change files outside the declared owned paths except docs/awf-plans generated by AWF planning.

---
Validation: 7 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: src/awf/service/metrics.py [hunk 1]
```diff
@@ -361,9 +374,76 @@ async def summarize_failure_analysis_for_session(
...
+        reason = row.failure_reason or UNKNOWN_FAILURE_REASON
+        msg = row.failure_message or ""
+
+        likely_cause = "Unknown Validation Failure"
+        action = "Review validation logs"
+
+        if "AGENT_AUTH_FAILED" in msg:
+            likely_cause = "Agent Auth Failed"
+            action = "Check agent credentials"
+        elif "model not found" in msg or "404" in msg:
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py:416]: fixed in commit 2bbac78

File: src/awf/service/metrics.py [hunk 2]
```diff
@@ -99,6 +99,17 @@ class FailedWorkspaceExample:
...
 
+@dataclass(frozen=True)
+class RootCauseCluster:
+    agent: str
+    agent_model: str | None
+    failure_reason: str
+    likely_cause: str
+    actionable_next_action: str
+    count: int
+    sample_workspace_ids: list[str]
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py]: fixed in commit f5d60040e587710296e833d47d8d31441011edd3

File: src/awf/service/metrics.py [hunk 3]
```diff
@@ -361,9 +374,76 @@ async def summarize_failure_analysis_for_session(
...
+            agent=k[0],
+            agent_model=k[1],
+            failure_reason=k[2],
+            likely_cause=k[3],
+            actionable_next_action=k[4],
+            count=len(wids),
+            sample_workspace_ids=wids[:5],
+        )
+        for k, wids in clusters.items()
+    ]
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py:446]: fixed in commit 09949ff

File: src/awf/service/metrics.py [hunk 4]
```diff
@@ -361,9 +374,76 @@ async def summarize_failure_analysis_for_session(
...
+
+    return [
+        RootCauseCluster(
+            agent=k[0],
+            agent_model=k[1],
+            failure_reason=k[2],
+            likely_cause=k[3],
+            actionable_next_action=k[4],
+            count=len(wids),
+            sample_workspace_ids=wids[:5],
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py]: fixed in commit 5b7eafe05e880153a636b5d7016ace2167f22e45

File: src/awf/service/metrics.py [hunk 5]
```diff
@@ -361,9 +374,76 @@ async def summarize_failure_analysis_for_session(
...
+            action = "Update tests to improve coverage or adjust baseline"
+        elif "SyntaxError" in msg or "ImportError" in msg:
+            likely_cause = "Syntax or Import Error"
+            action = "Fix syntax/import issues in generated code"
+        elif "GitHub auth/PR creation failed" in msg:
+            likely_cause = "GitHub Transient/Auth Error"
+            action = "Check GitHub App token or retry"
+        elif reason == FailureReason.agent_failure.value:
+            likely_cause = "Unknown Agent Failure"
+            action = "Review agent logs"
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py:430]: FALSE POSITIVE: Adding a return statement inside the loop would prematurely exit the function and break the clustering logic, as the `action` variable is merely a string used to form a dictionary key for the current row.

File: src/awf/service/metrics.py [hunk 6]
```diff
@@ -361,9 +374,76 @@ async def summarize_failure_analysis_for_session(
...
+
+        key = (agent, agent_model, reason, likely_cause, action)
+        if key not in clusters:
+            clusters[key] = []
+        clusters[key].append(row.id)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py]: fixed in commit 771e4d2758809c7b424d86a5beef23bb1327f525

File: src/awf/service/metrics.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/metrics.py]: fixed in commit cd11ada66019a3a09edc4243e472b9320746149f
